`hone-maven-plugin:0.23.3` doesn't work with Java 17

I'm trying to build the [objectionary/kafka](https://github.com/objectionary/kafka) with the following `hone-maven-plugin` configuration:

```gradle
def v = '0.23.3'
def hone = tasks.register('hone', Exec) {
  group = 'build'
  description = 'Run hone after Java compile'
  sourceSets.main.output.classesDirs.files.each { dir ->
    commandLine 'mvn',
            "-Dhone.timeout=700",
            "-Dhone.threads=0",
            "-Dhone.jeo-version=0.14.11",
            "-Dhone.grep-in=(66-69-6C-74-65-72|6D-61-70)",
            "-Dhone.rules=streams/*",
            "-Dhone.target=${project.buildDir}",
            "-Dhone.classes=${project.buildDir.toPath().relativize(dir.toPath())}",
            "org.eolang:hone-maven-plugin:${v}:pull",
            "org.eolang:hone-maven-plugin:${v}:optimize"
  }
}
```

and get the following error:

```
[WARNING] Error injecting: org.eolang.hone.PullMojo
java.lang.UnsupportedClassVersionError: org/cactoos/Input has been compiled by a more recent version of the Java Runtime (class file version 63.0), this version of the Java Runtime only recognizes class file versions up to 61.0
    at java.lang.ClassLoader.defineClass1 (Native Method)
    at java.lang.ClassLoader.defineClass (ClassLoader.java:1017)
    at java.security.SecureClassLoader.defineClass (SecureClassLoader.java:150)
    at java.net.URLClassLoader.defineClass (URLClassLoader.java:524)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:427)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:421)
    at java.security.AccessController.doPrivileged (AccessController.java:712)
    at java.net.URLClassLoader.findClass (URLClassLoader.java:420)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.findClassInternal (ClassRealm.java:256)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClassFromSelf (ClassRealm.java:351)
    at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass (SelfFirstStrategy.java:36)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass (ClassRealm.java:225)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:210)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:205)
    at java.lang.Class.getDeclaredConstructors0 (Native Method)
    at java.lang.Class.privateGetDeclaredConstructors (Class.java:3373)
    at java.lang.Class.getDeclaredConstructors (Class.java:2555)
    at com.google.inject.spi.InjectionPoint.forConstructorOf (InjectionPoint.java:299)
    at com.google.inject.internal.ConstructorBindingImpl.create (ConstructorBindingImpl.java:121)
    at com.google.inject.internal.InjectorImpl.createUninitializedBinding (InjectorImpl.java:715)
    at com.google.inject.internal.InjectorImpl.createJustInTimeBinding (InjectorImpl.java:941)
    at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecursive (InjectorImpl.java:863)
    at com.google.inject.internal.InjectorImpl.getJustInTimeBinding (InjectorImpl.java:300)
    at com.google.inject.internal.InjectorImpl.getBindingOrThrow (InjectorImpl.java:223)
    at com.google.inject.internal.InjectorImpl.getProviderOrThrow (InjectorImpl.java:1093)
    at com.google.inject.internal.InjectorImpl.getProvider (InjectorImpl.java:1122)
    at com.google.inject.internal.InjectorImpl.getProvider (InjectorImpl.java:1087)
    at com.google.inject.internal.InjectorImpl.getInstance (InjectorImpl.java:1139)
    at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:50)
    at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:86)
    at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision (InternalFactoryToInitializableAdapter.java:57)
    at com.google.inject.internal.ProviderInternalFactory$1.call (ProviderInternalFactory.java:67)
    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:109)
    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:124)
    at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)
    at com.google.inject.internal.ProviderInternalFactory.circularGet (ProviderInternalFactory.java:62)
    at com.google.inject.internal.InternalFactoryToInitializableAdapter.get (InternalFactoryToInitializableAdapter.java:47)
    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1101)
    at org.eclipse.sisu.inject.Guice4$2.get (Guice4.java:235)
    at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:83)
    at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:53)
    at org.codehaus.plexus.DefaultPlexusContainer.lookup (DefaultPlexusContainer.java:267)
    at org.codehaus.plexus.DefaultPlexusContainer.lookup (DefaultPlexusContainer.java:259)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo (DefaultMavenPluginManager.java:491)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:114)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.303 s
[INFO] Finished at: 2026-04-21T08:30:27+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eolang:hone-maven-plugin:0.23.3:pull (default-cli) on project standalone-pom: Execution default-cli of goal org.eolang:hone-maven-plugin:0.23.3:pull failed: An API incompatibility was encountered while executing org.eolang:hone-maven-plugin:0.23.3:pull: java.lang.UnsupportedClassVersionError: org/cactoos/Input has been compiled by a more recent version of the Java Runtime (class file version 63.0), this version of the Java Runtime only recognizes class file versions up to 61.0
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.eolang:hone-maven-plugin:0.23.3
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/.m2/repository/org/eolang/hone-maven-plugin/0.23.3/hone-maven-plugin-0.23.3.jar
[ERROR] urls[1] = file:/Users/.m2/repository/com/yegor256/jaxec/0.5.1/jaxec-0.5.1.jar
[ERROR] urls[2] = file:/Users/.m2/repository/com/jcabi/jcabi-log/0.24.3/jcabi-log-0.24.3.jar
[ERROR] urls[3] = file:/Users/.m2/repository/io/github/classgraph/classgraph/4.8.184/classgraph-4.8.184.jar
[ERROR] urls[4] = file:/Users/.m2/repository/org/apache/commons/commons-csv/1.14.1/commons-csv-1.14.1.jar
[ERROR] urls[5] = file:/Users/.m2/repository/commons-io/commons-io/2.21.0/commons-io-2.21.0.jar
[ERROR] urls[6] = file:/Users/.m2/repository/commons-codec/commons-codec/1.21.0/commons-codec-1.21.0.jar
[ERROR] urls[7] = file:/Users/.m2/repository/ch/qos/reload4j/reload4j/1.2.26/reload4j-1.2.26.jar
[ERROR] urls[8] = file:/Users/.m2/repository/com/jcabi/jcabi-maven-slf4j/0.12.2/jcabi-maven-slf4j-0.12.2.jar
[ERROR] urls[9] = file:/Users/.m2/repository/net/java/dev/jna/jna/5.18.1/jna-5.18.1.jar
[ERROR] urls[10] = file:/Users/.m2/repository/org/cactoos/cactoos/0.59.0/cactoos-0.59.0.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR]
[ERROR] -----------------------------------------------------
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException

[Incubating] Problems report is available at: file:///Users/lombrozo/Workspace/OpenSource/kafka-hone-opt/build/reports/problems/problems-report.html
```

`objectionary/kafka` requires Java 17 installed:

> You need **Java 17** — exactly this version, not less, not more!

--- a/.github/workflows/mvn.yml
+++ b/.github/workflows/mvn.yml
@@ -17,7 +17,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-24.04 ]
-        java: [ 22 ]
+        java: [ 17, 22 ]
     runs-on: ${{ matrix.os }}
     env:
       LC_ALL: C

